samples/fxa/src/main/java/org/mozilla/samples/fxa/MainActivity.kt

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.samples.fxa
+
+import android.net.Uri
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import mozilla.components.fxa.FirefoxAccount
+import android.support.customtabs.CustomTabsIntent
+import android.view.View
+import android.widget.Button
+import mozilla.components.fxa.Config
+import mozilla.components.fxa.FxaClient
+import android.content.Intent
+import android.widget.TextView
+
+open class MainActivity : AppCompatActivity() {
+
+    private var account: FirefoxAccount? = null
+    private var config: Config? = null
+
+    companion object {
+        const val CLIENT_ID = "12cc4070a481bc73"
+        const val REDIRECT_URL = "fxaclient://android.redirect"
+        const val CONFIG_URL = "https://latest.dev.lcip.org"
+
+        init {
+            FxaClient.init()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        config = Config.custom(CONFIG_URL)
+        config?.let {
+            account = FirefoxAccount(it, CLIENT_ID)
+
+            val btn = findViewById<View>(R.id.button) as Button
+            btn.setOnClickListener {
+                account?.beginOAuthFlow(REDIRECT_URL, arrayOf("profile"), false)?.let {
+                    openAuthTab(it)
+                }
+            }
```

Comment:
I agree with your issue about making this asynchronously. This synchronous chain of nullables is not fun :)

Reply:
👍